base/ServerManager.F90

```diff
@@ -159,14 +159,19 @@ subroutine initialize(this, comm, unusable, application_size, nodes_input_server
             npes_in(1)  == 0 .and. nodes_in(1)  == 0) profiler_name = "io_server_client"
 
         if (npes_in(1) == 0 .and. nodes_in(1) == 0) then
-           allocate(this%i_server, source = MpiServer(client_comm, 'i_server'//trim(i_to_string(1)),profiler_name=profiler_name, with_profiler=with_profiler))
+           allocate(this%i_server, source = MpiServer(client_comm, 'i_server'//trim(i_to_string(1)),profiler_name=profiler_name, &
+                                                      with_profiler=with_profiler, rc=status))
+           _VERIFY(status)
```

Comment:
These might as well go to `__RC__` as well. I think there are a few others here.

Reply:
There are a few others, but not this one.

Reply:
Ohhh. Duh. I didn't see this was inside an `allocate()`. I just saw `rc=status` and my brain went into overdrive. 😄

Reply:
Ah - we have to be careful with these.   The expression with `rc=status` is itself inside an `ALLOCATE` statement, so there are potentially two different failures.    Ordinarily I prefer not to worry about insufficient memory when allocating, but in these cases, if the expression for SOURCE fails, then the ALLOCATE fails for more immediate reasons.    So we should check both.     

Further, we need to check the expression result first as it is the more meaningful one.    So we need two different status's and can't use the macros at all.
```suggestion
           allocate(this%i_server, source = MpiServer(client_comm, 'i_server'//trim(i_to_string(1)),profiler_name=profiler_name, &
                                                      with_profiler=with_profiler, rc=status), stat=status_alloc)
           _VERIFY(status)
           _VERIFY(status_alloc)
```
You'll need to add the new integer `status_alloc` in the top of the procedure.  (Not wed to this name.)

Reply:
I'm unresolving this. I think I resolved it just about when @tclune commented, so it's possible @weiyuan-jiang never saw this!.

But as @tclune says, it's probably good to test both returns here!